commit 7b3d0777ea6c10dd430ed686434d3a4e2d072632
Author: Carson Black <uhhadd@gmail.com>
Date:   Wed Aug 28 17:21:15 2019 -0400

    [GTK3] Have checkboxes and radiobuttons respect the user's color scheme.
    
    Summary: Checkboxes and radiobuttons are now recolored according to the user's colour tastes.
    
    Test Plan:
    Old: {F7290945, size=full}
    
    New: {F7290947, size=full}
    
    Reviewers: #breeze, ndavis
    
    Reviewed By: #breeze, ndavis
    
    Subscribers: ndavis, ngraham, plasma-devel
    
    Tags: #plasma
    
    Differential Revision: https://phabricator.kde.org/D23547

diff --git b/src/assets/breeze-check-checked-symbolic.svg a/src/assets/breeze-check-checked-symbolic.svg
deleted file mode 100644
index 845e055..0000000
--- b/src/assets/breeze-check-checked-symbolic.svg
+++ /dev/null
@@ -1,6 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<svg id="svg8" version="1.1" viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg">
-    <rect id="rect2" class="warning" x="1.5" y="1.5" width="15" height="15" rx="2" ry="2" fill="#eff0f1"/>
-    <path id="path4" class="error" d="m4 4v10h10v-10z" fill="#3daee9"/>
-    <path id="path6" class="error" d="m3.5 1c-1.385 0-2.5 1.115-2.5 2.5v11c0 1.385 1.115 2.5 2.5 2.5h11c1.385 0 2.5-1.115 2.5-2.5v-11c0-1.385-1.115-2.5-2.5-2.5zm0 1h11c0.831 0 1.5 0.669 1.5 1.5v11c0 0.831-0.669 1.5-1.5 1.5h-11c-0.831 0-1.5-0.669-1.5-1.5v-11c0-0.831 0.669-1.5 1.5-1.5z" color="#000000" color-rendering="auto" dominant-baseline="auto" fill="#7c8288" image-rendering="auto" shape-rendering="auto" solid-color="#000000" style="font-feature-settings:normal;font-variant-alternates:normal;font-variant-caps:normal;font-variant-ligatures:normal;font-variant-numeric:normal;font-variant-position:normal;isolation:auto;mix-blend-mode:normal;shape-padding:0;text-decoration-color:#000000;text-decoration-line:none;text-decoration-style:solid;text-indent:0;text-orientation:mixed;text-transform:none;white-space:normal"/>
-</svg>
diff --git b/src/assets/breeze-check-indeterminate-symbolic.svg a/src/assets/breeze-check-indeterminate-symbolic.svg
deleted file mode 100644
index 4fac79f..0000000
--- b/src/assets/breeze-check-indeterminate-symbolic.svg
+++ /dev/null
@@ -1,6 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<svg version="1.1" viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg">
-    <rect class="warning" x="1.5" y="1.5" width="15" height="15" rx="2" ry="2" fill="#eff0f1"/>
-    <path class="error" d="m4 4v10h10v-10zm7.293 2h0.70703v6h-6v-0.70703z" fill="#3daee9"/>
-    <path class="error" d="m3.5 1c-1.385 0-2.5 1.115-2.5 2.5v11c0 1.385 1.115 2.5 2.5 2.5h11c1.385 0 2.5-1.115 2.5-2.5v-11c0-1.385-1.115-2.5-2.5-2.5zm0 1h11c0.831 0 1.5 0.669 1.5 1.5v11c0 0.831-0.669 1.5-1.5 1.5h-11c-0.831 0-1.5-0.669-1.5-1.5v-11c0-0.831 0.669-1.5 1.5-1.5z" color="#000000" color-rendering="auto" dominant-baseline="auto" fill="#7c8288" image-rendering="auto" shape-rendering="auto" solid-color="#000000" style="font-feature-settings:normal;font-variant-alternates:normal;font-variant-caps:normal;font-variant-ligatures:normal;font-variant-numeric:normal;font-variant-position:normal;isolation:auto;mix-blend-mode:normal;shape-padding:0;text-decoration-color:#000000;text-decoration-line:none;text-decoration-style:solid;text-indent:0;text-orientation:mixed;text-transform:none;white-space:normal"/>
-</svg>
diff --git b/src/assets/breeze-check-unchecked-symbolic.svg a/src/assets/breeze-check-unchecked-symbolic.svg
deleted file mode 100644
index 0b08747..0000000
--- b/src/assets/breeze-check-unchecked-symbolic.svg
+++ /dev/null
@@ -1,5 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<svg id="svg8" version="1.1" viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg">
-    <rect id="rect2" class="warning" x="1.5" y="1.5" width="15" height="15" rx="2" ry="2" fill="#eff0f1"/>
-    <path id="path6" class="success" d="m3.5 1c-1.385 0-2.5 1.115-2.5 2.5v11c0 1.385 1.115 2.5 2.5 2.5h11c1.385 0 2.5-1.115 2.5-2.5v-11c0-1.385-1.115-2.5-2.5-2.5zm0 1h11c0.831 0 1.5 0.669 1.5 1.5v11c0 0.831-0.669 1.5-1.5 1.5h-11c-0.831 0-1.5-0.669-1.5-1.5v-11c0-0.831 0.669-1.5 1.5-1.5z" color="#000000" color-rendering="auto" dominant-baseline="auto" fill="#7c8288" image-rendering="auto" shape-rendering="auto" solid-color="#000000" style="font-feature-settings:normal;font-variant-alternates:normal;font-variant-caps:normal;font-variant-ligatures:normal;font-variant-numeric:normal;font-variant-position:normal;isolation:auto;mix-blend-mode:normal;shape-padding:0;text-decoration-color:#000000;text-decoration-line:none;text-decoration-style:solid;text-indent:0;text-orientation:mixed;text-transform:none;white-space:normal"/>
-</svg>
diff --git b/src/assets/breeze-radio-checked-symbolic.svg a/src/assets/breeze-radio-checked-symbolic.svg
deleted file mode 100644
index b8db473..0000000
--- b/src/assets/breeze-radio-checked-symbolic.svg
+++ /dev/null
@@ -1,6 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<svg id="svg8" version="1.1" viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg">
-    <path id="path2" class="warning" d="M 16.5,9 A 7.5,7.5 0 0 1 9,16.5 7.5,7.5 0 0 1 1.5,9 7.5,7.5 0 0 1 9,1.5 7.5,7.5 0 0 1 16.5,9 Z" fill="#f57900"/>
-    <path id="path4" class="error" d="m9 1a8 8 0 0 0-8 8 8 8 0 0 0 8 8 8 8 0 0 0 8-8 8 8 0 0 0-8-8zm0 1a7 7 0 0 1 7 7 7 7 0 0 1-7 7 7 7 0 0 1-7-7 7 7 0 0 1 7-7z" fill="#ff0404"/>
-    <path id="path6" class="error" d="m9 4c-2.7614 0-5 2.2386-5 5 0 2.7614 2.2386 5 5 5 2.7614 0 5-2.2386 5-5 0-2.7614-2.2386-5-5-5z" color="#000000" color-rendering="auto" dominant-baseline="auto" fill="#ff0404" image-rendering="auto" shape-rendering="auto" solid-color="#000000" stroke-width="3.9903" style="font-feature-settings:normal;font-variant-alternates:normal;font-variant-caps:normal;font-variant-ligatures:normal;font-variant-numeric:normal;font-variant-position:normal;isolation:auto;mix-blend-mode:normal;shape-padding:0;text-decoration-color:#000000;text-decoration-line:none;text-decoration-style:solid;text-indent:0;text-orientation:mixed;text-transform:none;white-space:normal"/>
-</svg>
diff --git b/src/assets/breeze-radio-indeterminate-symbolic.svg a/src/assets/breeze-radio-indeterminate-symbolic.svg
deleted file mode 100644
index 62f610b..0000000
--- b/src/assets/breeze-radio-indeterminate-symbolic.svg
+++ /dev/null
@@ -1,6 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<svg version="1.1" viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg">
-    <path class="warning" d="M 16.5,9 A 7.5,7.5 0 0 1 9,16.5 7.5,7.5 0 0 1 1.5,9 7.5,7.5 0 0 1 9,1.5 7.5,7.5 0 0 1 16.5,9 Z" fill="#f57900"/>
-    <path class="error" d="m9 1a8 8 0 0 0-8 8 8 8 0 0 0 8 8 8 8 0 0 0 8-8 8 8 0 0 0-8-8zm0 1a7 7 0 0 1 7 7 7 7 0 0 1-7 7 7 7 0 0 1-7-7 7 7 0 0 1 7-7z" fill="#ff0404"/>
-    <path class="error" d="m9 4a5 5 0 0 0-5 5 5 5 0 0 0 5 5 5 5 0 0 0 5-5 5 5 0 0 0-5-5zm1.75 2.8066a0.66 0.66 0 0 1 0.47656 0.19336 0.66 0.66 0 0 1 0.080078 0.097656 3 3 0 0 1 0.69336 1.9023 3 3 0 0 1-3 3 3 3 0 0 1-1.9512-0.72852 3 3 0 0 1-0.048828-0.042968v-2e-3a0.66 0.66 0 0 1 0-0.93359l3.293-3.2929a0.66 0.66 0 0 1 0.45703-0.19336z" color="#000000" color-rendering="auto" dominant-baseline="auto" fill="#ff0404" image-rendering="auto" shape-rendering="auto" solid-color="#000000" stroke-width="3.9903" style="font-feature-settings:normal;font-variant-alternates:normal;font-variant-caps:normal;font-variant-ligatures:normal;font-variant-numeric:normal;font-variant-position:normal;isolation:auto;mix-blend-mode:normal;shape-padding:0;text-decoration-color:#000000;text-decoration-line:none;text-decoration-style:solid;text-indent:0;text-orientation:mixed;text-transform:none;white-space:normal"/>
-</svg>
diff --git b/src/assets/breeze-radio-unchecked-symbolic.svg a/src/assets/breeze-radio-unchecked-symbolic.svg
deleted file mode 100644
index 28aaf1b..0000000
--- b/src/assets/breeze-radio-unchecked-symbolic.svg
+++ /dev/null
@@ -1,5 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<svg id="svg8" version="1.1" viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg">
-    <path id="path2" class="warning" d="M 16.5,9 A 7.5,7.5 0 0 1 9,16.5 7.5,7.5 0 0 1 1.5,9 7.5,7.5 0 0 1 9,1.5 7.5,7.5 0 0 1 16.5,9 Z" fill="#f57900"/>
-    <path id="path4" class="success" d="m9 1a8 8 0 0 0-8 8 8 8 0 0 0 8 8 8 8 0 0 0 8-8 8 8 0 0 0-8-8zm0 1a7 7 0 0 1 7 7 7 7 0 0 1-7 7 7 7 0 0 1-7-7 7 7 0 0 1 7-7z" fill="#ff0404"/>
-</svg>
diff --git b/src/gtk3/widgets/_checkboxes.scss a/src/gtk3/widgets/_checkboxes.scss
index b9fa4bd..f6eba2a 100644
--- b/src/gtk3/widgets/_checkboxes.scss
+++ a/src/gtk3/widgets/_checkboxes.scss
@@ -3,6 +3,20 @@
 /*************************
  * Check and Radio items *
  *************************/
+@each $s,$as in ('','-unchecked'),
+                 (':hover', '-unchecked-hover'),
+                 (':active', '-unchecked-active'),
+                 (':backdrop', '-unchecked-backdrop'),
+                 (':checked', '-checked-active'),
+                 (':checked:hover', '-checked-hover'),
+                 (':checked:active', '-checked-active'),
+                 (':backdrop:checked', '-checked-backdrop') {
+ .view.content-view.check#{$s}:not(list) {
+   -gtk-icon-shadow: none;
+   -gtk-icon-source: -gtk-scaled(url("../assets/check-selectionmode#{$as}.png"), url("../assets/check-selectionmode#{$as}@2.png"));
+   background-color: transparent;
+ }
+}
 
 checkbutton.text-button, radiobutton.text-button {
  // this is for a nice focus on check and radios text
@@ -18,43 +32,53 @@ checkbutton.text-button, radiobutton.text-button {
  }
 }
 
-check, radio {
-  margin: 0 4px;
+@each $a in ('check','radio') {
 
-  &:only-child { margin: 0; }
+  #{$a} {
+   margin: 0 4px;
 
-  min-height: 18px;
-  min-width: 18px;
-  //border: none;
-  animation: none;
-  background-color: transparent;
-  -gtk-icon-palette: error gtk("@theme_button_decoration_focus"), success gtk("@borders"), warning gtk("@theme_button_background_normal");
-  -gtk-icon-shadow: 1px 1px rgba(35,38,41,0.2);
+   &:only-child { margin: 0; }
 
-  &:hover {
-    -gtk-icon-palette: error gtk("@theme_button_decoration_focus"), success gtk("@theme_button_decoration_focus"), warning gtk("@theme_button_background_normal");
-  }
-  &:disabled {
-    -gtk-icon-palette: error gtk("@theme_button_decoration_focus_insensitive"), success gtk("@insensitive_borders"), warning gtk("@theme_button_background_insensitive");
-  }
+   min-height: 18px;
+   min-width: 18px;
+   //border: none;
+   animation: none;
+   background-color: transparent;
+   -gtk-icon-source: -gtk-scaled(url("../assets/#{$a}-unchecked.png"), url("../assets/#{$a}-unchecked@2.png"));
 
-  menu menuitem & {
-    margin: 0; // this is a workaround for a menu check/radio size allocation issue
-    min-height: 18px;
-    min-width: 18px;
-    background-color: transparent;
-    box-shadow: none;
-    -gtk-icon-shadow: none;
-    animation: none;
-  }
-}
-@each $type in ('check', 'radio') {
-  #{$type} {
-    -gtk-icon-source: -gtk-recolor(url("../assets/breeze-#{$type}-unchecked-symbolic.svg"));
-    @each $state in ('checked', 'indeterminate') {
-      &:#{$state} {
-        -gtk-icon-source: -gtk-recolor(url("../assets/breeze-#{$type}-#{$state}-symbolic.svg"));
-      }
+   @each $s,$as in ('hover','-hover'),
+                    ('active','-active'),
+                    ('disabled','-insensitive'),
+                    ('backdrop','-backdrop'),
+                    ('backdrop:disabled','-backdrop-insensitive') {
+    &:#{$s} { -gtk-icon-source: -gtk-scaled(url("../assets/#{$a}-unchecked#{$as}.png"), url("../assets/#{$a}-unchecked#{$as}@2.png")); }
+    &:checked {
+      -gtk-icon-source: -gtk-scaled(url("../assets/#{$a}-checked-active.png"), url("../assets/#{$a}-checked-active@2.png"));
+      &:#{$s} { -gtk-icon-source: -gtk-scaled(url("../assets/#{$a}-checked#{$as}.png"), url("../assets/#{$a}-checked#{$as}@2.png")); }
     }
+    &:indeterminate {
+      -gtk-icon-source: -gtk-scaled(url("../assets/#{$a}-mixed-active.png"), url("../assets/#{$a}-mixed-active@2.png"));
+      &:#{$s} { -gtk-icon-source: -gtk-scaled(url("../assets/#{$a}-mixed#{$as}.png"), url("../assets/#{$a}-mixed#{$as}@2.png")); }
+    }
+  }
+  
+   menu menuitem & {
+     margin: 0; // this is a workaround for a menu check/radio size allocation issue
+     min-height: 18px;
+     min-width: 18px;
+     background-color: transparent;
+     box-shadow: none;
+     -gtk-icon-shadow: none;
+     animation: none;
+     &:hover {
+       -gtk-icon-source: -gtk-scaled(url("../assets/#{$a}-unchecked.png"), url("../assets/#{$a}-unchecked@2.png"));
+     }
+     &:checked:hover {
+       -gtk-icon-source: -gtk-scaled(url("../assets/#{$a}-checked-active.png"), url("../assets/#{$a}-checked-active@2.png"));
+     }
+     &:indeterminate:hover {
+       -gtk-icon-source: -gtk-scaled(url("../assets/#{$a}-mixed-active.png"), url("../assets/#{$a}-mixed-active@2.png"));
+     }
+   }
   }
 }
